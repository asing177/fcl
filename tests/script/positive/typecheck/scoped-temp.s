transition initial -> terminal;

@initial
foo () {
  if (true) {
    tmp = true
  } else {
    tmp = "bla" // ok, the other `tmp` is not in scope
  };
  transitionTo(@terminal)
}

@initial
bar() {
  after (now()) {
    tmp = true
  };
  tmp = 1; // also ok
  transitionTo(@terminal)
}
