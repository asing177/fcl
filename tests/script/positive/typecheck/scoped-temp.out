 
Method arguments and effects:
foo: () -> {}
bar: () -> {read via now}